### PR TITLE
Native image - make it possible to generate the dashboard dump file

### PR DESCRIPTION
**Pull request by mkouba**
See also https://www.graalvm.org/docs/tools/dashboard/ and https://groups.google.com/g/quarkus-dev/c/f9d5lMQWGFQ

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
Nice, thanks!

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java [hunk 1]
```diff
@@ -336,6 +336,12 @@
         public boolean enabled;
     }
 
+    /**
+     * Generate the report files for GraalVM Dashboard.
+     */
+    @ConfigItem
+    public boolean enableDashboardDump;
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:343]: Should we enable it for one of our ITs in the tree? I don't know if it slows down things a lot or requires a ton of memory?

Comment (mkouba) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:343]: The dump file has 112M for a simple app. Do we enable reports (used_classes, etc.) for some ITs?

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:343]: OK, let's merge as is then. Looks a bit too big for our own good.

**Pull request merged**